src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -355,14 +356,26 @@ private IEventReferenceExpression CreateBoundEventAccessOperation(BoundEventAcce
 
         private IEventAssignmentExpression CreateBoundEventAssignmentOperatorOperation(BoundEventAssignmentOperator boundEventAssignmentOperator)
         {
-            IEventSymbol @event = boundEventAssignmentOperator.Event;
-            Lazy<IOperation> eventInstance = new Lazy<IOperation>(() => Create(boundEventAssignmentOperator.Event.IsStatic ? null : boundEventAssignmentOperator.ReceiverOpt));
+            SyntaxNode syntax = boundEventAssignmentOperator.Syntax;
+            Lazy<IEventReferenceExpression> eventReference = new Lazy<IEventReferenceExpression>(() =>
+            {
+                // BoundEventAssignmentOperator doesn't hold on to BoundEventAccess provided during binding.
+                // Based on the implementation of those two bound node types, the following data can be retrieved w/o changing BoundEventAssignmentOperator:
+                //  1. the type of BoundEventAccess is the type of the event symbol.
+                //  2. the constant value of BoundEventAccess is always null.
+                //  3. the syntax of the boundEventAssignmentOperator is always AssignmentExpressionSyntax, so the syntax for the event reference would be the LHS of the assignment.
+                IEventSymbol @event = boundEventAssignmentOperator.Event;
+                Lazy<IOperation> instance = new Lazy<IOperation>(() => Create(boundEventAssignmentOperator.Event.IsStatic ? null : boundEventAssignmentOperator.ReceiverOpt));
```

Comment:
As I just mentioned offline, making this change causes us to return a `None` operation as the instance derived from a `BoundTypeExpression` as `BoundEventAssignmentOperator.ReceiverOpt`. If we want to expose as much information as possible in erroneous case, we will need to accurately check for error when creating the `EventReferenceOperation` plus modify the interface definition to make it clear that instance could be non null in this case. We probably discuss this during our design meeting.

Meanwhile, I 'm gonna revert this change (but keep those tests) to make the behavior consistent with other related operations, e.g.:
https://github.com/dotnet/roslyn/blob/features/ioperation/src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs#L306

Is this OK?